src/huggingface_hub/hf_api.py

```diff
@@ -506,6 +506,85 @@ def list_models(
         d = r.json()
         return [ModelInfo(**x) for x in d]
 
+    def list_models_from_filter(self, model_filter: ModelFilter):
+        """
+        Get the public list of all the models on huggingface.co based on a `ModelFilter`
+        Args:
+            model_filter (:obj: `ModelFilter`):
+                An Enum class which handles storing of query information
+                Example usage:
+
+                    >>> from huggingface_hub import HfApi, ModelFilter
+                    >>> api = HfApi()
+
+                    >>> # List only the text classification models
+                    >>> f = ModelFilter(task="text-classification")
+                    >>> api.list_models_from_filter(filter=f)
```

Comment:
I think the `ModelFilter` class has good potential to be helpful, but right now it doesn't feel very different from the `list_models` method.

Doing the following doesn't seem necessarily simpler
```
>>> f = ModelFilter(task="text-classification")
>>> api.list_models_from_filter(filter=f)
```

than doing this
```
>>> api.list_models(filter="text-classification")
```

Reply:
I think it has very good potential to be super useful by leveraging your PR here https://github.com/huggingface/huggingface_hub/pull/537, i.e., offering auto-completion, or am I missing something?

Reply:
What if you'd like to filter with several tasks, how would you handle that? I think offering multiple selectors (language, task) per `ModelFilter` makes sense, but then I'd also expect the ability to offer multiple selectors from the same category

Reply:
Threw that into https://github.com/huggingface/huggingface_hub/pull/556

IMO: This PR stays a bit dormant until all of the other necessary bits are needed to *only* add in a new filter API function, and this would be for complex queries ideally.